core/tools.js

```diff
@@ -1569,6 +1569,19 @@
 			return base64string;
 		},
 
+		/**
+		 * Counts childes of given {@link CKEDITOR.dom.element} omitting {@link CKEDITOR.dom.comment}.
+		 *
+		 * @since 4.11.2
+		 * @param {CKEDITOR.dom.element} node Element whos children will be counted.
+		 * @returns {Number} Amount of node's children without counting comments.
+		 */
+		childCountWithoutComments: function( node ) {
```

Comment:
We can't add a method like that, it doesn't belong here.

1. Logically it belongs to `CKEDITOR.dom.element` type - this is where it should be added.
1. It's too specific to your particular use case. Having a method like that would be excuse for getChildCountWithoutNodes, getChildCountWithoutElements etc. Instead [`getChildCount` method](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_dom_element.html#method-getChildCount) should be modified to accept `options` param, that allows to specify `options.exclude` or `options.include`. So you'd call it like `elem.getChildCount( { exclude: [ CKEDITOR.NODE_COMMENT ] } );`.
1. Name is not aligned with existing [`getChildCount` method](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_dom_element.html#method-getChildCount).

Finally there's a minor naming issue, `node` can not have children so the arg should be called `element` just like you described in `@param`'s description.

I should have pointed this out in previous review.